eudemo/eudemo/maintenance/lower_port/designer.py

```diff
@@ -0,0 +1,595 @@
+        z_highest = max(c_pts, key=lambda p: p[2])
+
+        return (z_highest[0], z_highest[2]), (div_x_outer, div_z_top)
+
+    def _pad_points(
+        self,
+        inner_point: Tuple,
+        outer_point: Tuple,
+    ):
+        points_grad = (inner_point[1] - outer_point[1]) / (
+            inner_point[0] - outer_point[0]
+        )  # z/x
+        points_len, _ = distance_to(
+            [inner_point[0], 0, inner_point[1]],
+            [outer_point[0], 0, outer_point[1]],
+        )
+
+        (padded_bot_inner_point, _) = LowerPortDesigner._xz_points_dist_away_from(
+            outer_point,
+            points_grad,
+            points_len + self.lower_duct_div_padding / 2,
+        )
+        (
+            _,
+            padded_top_outer_point,
+        ) = LowerPortDesigner._xz_points_dist_away_from(
+            inner_point,
+            points_grad,
+            points_len + self.lower_duct_div_padding / 2,
+        )
+        return (
+            padded_bot_inner_point,
+            padded_top_outer_point,
+        )
+
+    def _angled_duct_edge_wall_points(
+        self,
+        duct_w_wall_koz: BluemiraFace,
+        padded_bottom_inner_point: Tuple,
+        padded_top_outer_point: Tuple,
+    ):
+        wall_bot_inner_point = duct_w_wall_koz.vertexes.points[
+            duct_w_wall_koz.vertexes.argmin(
+                [
+                    padded_bottom_inner_point[0],
+                    0,
+                    padded_bottom_inner_point[1],
+                ]
+            )
+        ]
+        wall_top_outer_point = duct_w_wall_koz.vertexes.points[
+            duct_w_wall_koz.vertexes.argmin(
+                [
+                    padded_top_outer_point[0],
+                    0,
+                    padded_top_outer_point[1],
+                ]
+            )
+        ]
+
+        return (
+            wall_bot_inner_point,
+            wall_top_outer_point,
+        )
+
+    def _straight_duct_extrude_boundary(
+        self,
+        start_top_point: Tuple,
+        start_bot_point: Tuple,
+        angled_y_len: float,
+    ) -> BluemiraFace:
+        x_point = start_top_point[0]
+        y_size = angled_y_len + self.lower_duct_straight_y_offset
+
+        return make_polygon(
+            {
+                "x": np.array(
+                    [
+                        x_point,
+                        x_point,
+                        x_point,
+                        x_point,
+                    ]
+                ),
+                "y": np.array(
+                    [
+                        y_size,
+                        -y_size,
+                        -y_size,
+                        y_size,
+                    ]
+                ),
+                "z": np.array(
+                    [
+                        start_top_point[1],
+                        start_top_point[1],
+                        start_bot_point[1],
+                        start_bot_point[1],
+                    ]
```

Comment:
for readability it would maybe be better to create these arrays before this function

Reply:
Ah I think it's more readable this way :/